tests/test_generate_ics.py

```
@@ -425,6 +449,29 @@ def test_filename_subdivision(self):
+
+    def test_filename_output_suffix_does_not_append_extension(self):
+        with self.temp_cwd() as temp_dir:
+            with self.argv("US", "--years", "2025", "--output-suffix", "-personal"):
+                IcsGenerator().run()
+
+            self.assertTrue((temp_dir / "US_2025-personal").exists())
+            self.assertFalse((temp_dir / "US_2025-personal.ics").exists())
+
+    def test_filename_output_and_suffix_are_mutually_exclusive(self):
+        with self.argv("US", "--output", "calendar.ics", "--output-suffix", "-personal.ics"):
+            with self.assertRaises(SystemExit):
```

Ruff (0.15.18)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)